inference_levels/local_inferencers/local_inference.py

```diff
@@ -0,0 +1,35 @@
+import torch
+
+
+class BaseLocalInference(torch.nn.Module):
+    def __init__(self,embedding_size):
+        super().__init__()
+
+        self.energy_metrix = None
```

Comment:
does this have to be a field? If it is re-created in each forward call, then it is better to have it a local variable in that function.
Saving it to the object state might be confusing, since it implies that this field has a purpose in the future, for which it is stored, and it might cause bugs when you try to access this field at inappropriate times.